### PR TITLE
GPS Control Panel and GEO Scanner access checks

### PR DESCRIPTION
**Pull request by NCBPFluffyBear**
## Description
Added checks to make sure that the player attempting to access the GPS Control Panel and GEO Scanner has sufficient permissions to access them.

## Changes
- slimefun.inventory.bypass permission check
- Protection manager inventory access permission check
- Research check
- No access message

## Related Issues
N/A

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

**Review by TheBusyBiscuit — DISMISSED** [review]
Can't really see of any good reason why this would be useful since those aren't inventories and don't let the player do anything other than checking their own info but there's still a lot of changes to be made if this should really be a desirable change.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/geo/GEOScanner.java [hunk 1]
```diff
@@ -19,10 +22,22 @@ public GEOScanner(Category category, SlimefunItemStack item, RecipeType recipeTy
...
+
             Block b = e.getClickedBlock().get();
+            Player p = e.getPlayer();
 
-            e.cancel();
-            SlimefunPlugin.getGPSNetwork().getResourceManager().scan(e.getPlayer(), b, 0);
+            if (p.hasPermission("slimefun.inventory.bypass")
+                || (SlimefunPlugin.getProtectionManager().hasPermission(
+                p, b.getLocation(), ProtectableAction.ACCESS_INVENTORIES))
+                && Slimefun.hasUnlocked(e.getPlayer(), item, false)
```
[the remark this answers is not in the document]
Comment (NCBPFluffyBear) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/geo/GEOScanner.java:33]: The entire thing? I have tested it and it allows other people to access them/activate them

**Comment by NCBPFluffyBear** [comment]
> Can't really see of any good reason why this would be useful since those aren't inventories and don't let the player do anything other than checking their own info but there's still a lot of changes to be made if this should really be a desirable change.

I thought this too, but it was brought to my attention that people that had the proper GPS complexity could activate the GEO Scanners of other players. Normally wouldn't be too bad, but on skyblock, its a free boost

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/geo/GEOScanner.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/geo/GEOScanner.java:33]: The `hasUnlocked()` check is redundant, handlers aren't even called unless the Player has access to the item.

**Review by Sfiguz7 — APPROVED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/geo/GEOScanner.java [hunk 2]
```diff
@@ -1,6 +1,9 @@
 package io.github.thebusybiscuit.slimefun4.implementation.items.geo;
 
+import io.github.thebusybiscuit.cscorelib2.protection.ProtectableAction;
+import me.mrCookieSlime.Slimefun.api.Slimefun;
```
Comment (LinoxGH) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/geo/GEOScanner.java:4]: Misplaced imports.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/gps/GPSControlPanel.java [hunk 3]
```diff
@@ -1,5 +1,8 @@
 package io.github.thebusybiscuit.slimefun4.implementation.items.gps;
 
+import io.github.thebusybiscuit.cscorelib2.protection.ProtectableAction;
+import me.mrCookieSlime.Slimefun.api.Slimefun;
```
Comment (LinoxGH) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/gps/GPSControlPanel.java:4]: Misplaced imports.

**Review by LinoxGH — CHANGES_REQUESTED** [review]

**Review by LinoxGH — APPROVED** [review]

**Review by Sfiguz7 — APPROVED** [review]

**Pull request merged**